Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -346,6 +346,52 @@ - (instancetype)createSession:(NSDictionary *)launchOptions useCustomLaunchOptio
 //////////////////////////////////////////////////////////////
 
 // documented in IterableAPI.h
++ (IterableAPI *)sharedInstance
+{
+    if (sharedInstance == nil) {
+        LogError(@"[sharedInstance called before sharedInstanceWithApiKey");
+    }
+    return sharedInstance;
+}
+
+// documented in IterableAPI.h
++ (IterableAPI *)sharedInstanceWithApiKey:(NSString *)apiKey andEmail:(NSString *)email launchOptions:(NSDictionary *)launchOptions
+{
+    // threadsafe way to create a static singleton https://stackoverflow.com/questions/5720029/create-singleton-using-gcds-dispatch-once-in-objective-c
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[IterableAPI alloc] initWithApiKey:apiKey andEmail:email launchOptions:launchOptions];
+    });
+    return sharedInstance;
+}
+
+// documented in IterableAPI.h
++ (IterableAPI *)sharedInstanceWithApiKey:(NSString *)apiKey andUserId:(NSString *)userId launchOptions:(NSDictionary *)launchOptions
+{
+    // threadsafe way to create a static singleton https://stackoverflow.com/questions/5720029/create-singleton-using-gcds-dispatch-once-in-objective-c
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[IterableAPI alloc] initWithApiKey:apiKey andUserId:userId launchOptions:launchOptions];
+    });
+    return sharedInstance;
+}
+
+// documented in IterableAPI.h
++(void) getAndTrackDeeplink:(NSURL *)webpageURL callbackBlock:(ITEActionBlock)callbackBlock
+{
+    if ([webpageURL.absoluteString rangeOfString:ITBL_DEEPLINK_IDENTIFIER].location != NSNotFound) {
```

Comment:
I think your regex would need to include more than that, since there are optional params that the url could have after the initial id. http://links.iterable.com/a/0691e8dc1a0e4f2faa346663fbfb5997?_e=joneng%40iterable.com&_m=93125f33ba814b13a882358f8e0852e0